vault/vault_auth.go

```diff
@@ -0,0 +1,172 @@
+package vault
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/hashicorp/vault/api"
+	"github.com/pkg/errors"
+	log "github.com/sirupsen/logrus"
+)
+
+const (
+	DefaultTokenTTL  = 86400 // 1 day
```

Comment:
Should we use `time.Second` unit for the default values to specify the unit is `Second`? Similar to [here](https://github.com/Nitro/sidecar/blob/master/discovery/discovery.go#L11)

```go
DefaultTokenTTL = 86400 * time.Second // 1 day
TokenGracePeriod = 3600 * time.Second // 1 hour
```